eng/pipelines/templates/steps/build.yml

```diff
@@ -64,13 +64,17 @@ steps:
     parameters:
       ServiceDirectory: ${{parameters.ServiceDirectory}}
 
-  - ${{ each artifact in parameters.Artifacts }}:
-    - pwsh: |
-        New-Item -Type Directory -Name ${{artifact.name}} -Path $(Build.ArtifactStagingDirectory)
-        Copy-Item sdk/${{parameters.ServiceDirectory}}/**/${{artifact.name}}-*.tgz $(Build.ArtifactStagingDirectory)/${{artifact.name}}
-        Copy-Item sdk/${{parameters.ServiceDirectory}}/**/browser/${{artifact.name}}-*.zip $(Build.ArtifactStagingDirectory)/${{artifact.name}}
-        Copy-Item $(Build.ArtifactStagingDirectory)/Documentation/${{artifact.name}}.zip $(Build.ArtifactStagingDirectory)/${{artifact.name}}/documentation
-      displayName: 'Copy Package ${{artifact.name}}'
+  - pwsh: |
+      $artifacts = '${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json
+      foreach ($artifact in $artifacts)
+      {
+        Write-Host "Copying $($artifact.name) artifacts to $(Build.ArtifactStagingDirectory)/$($artifact.name)"
```

Comment:
I suggest capturing the artifact name into a variable like:
```
$artifactName = $artifact.name
```
Then using that in all the places below as it current is I don't think `$artifact.name` is actually doing what you want in those paths.